forest/sycamore/tests/test_functions.py

```diff
@@ -113,16 +125,140 @@ def test_survey_submits_with_no_submissions():
         TEST_DATA_DIR, "sample_intervention_data.json"
     )
     agg_data = aggregate_surveys_config(study_dir, survey_settings_path, "UTC")
-    interventions_dict = get_all_interventions_dict(interventions_path)
     ss_detail, ss_summary = survey_submits(
         survey_settings_path, "2021-12-01", "2021-12-30",
-        ["idr8gqdh"], agg_data, interventions_dict
+        ["idr8gqdh"], agg_data, interventions_path
     )
     assert ss_detail.shape[0] == 0
     assert ss_summary.shape[0] == 0
 
 
 def test_survey_submits_no_config():
+    # Ensure that survey_submits_no_config generates the same information
+    # regardless of whether the passed data was generated from a config file.
+    study_dir = os.path.join(TEST_DATA_DIR, "sample_dir")
+    survey_settings_path = os.path.join(
+        TEST_DATA_DIR, "sample_study_surveys_and_settings.json"
+    )
+    agg_data = aggregate_surveys_config(study_dir, survey_settings_path, "UTC")
+    submits_tbl = survey_submits_no_config(agg_data)
+    assert submits_tbl.shape[0] == 6
+
+
+def test_survey_submits_no_config_with_config_input():
     study_dir = os.path.join(TEST_DATA_DIR, "sample_dir")
-    submits_tbl = survey_submits_no_config(study_dir, "UTC")
-    assert submits_tbl.shape[0] == 2
+    agg_data = aggregate_surveys_no_config(study_dir, study_tz="UTC")
+    submits_tbl = survey_submits_no_config(agg_data)
+    assert submits_tbl.shape[0] == 6
+
+
+def test_read_one_answers_stream():
+    study_dir = os.path.join(TEST_DATA_DIR, "sample_dir")
+    df = read_one_answers_stream(study_dir, "idr8gqdh", "UTC")
+    assert len(df["Local time"].unique()) == 2
+
+
+def test_read_aggregate_answers_stream():
+    study_dir = os.path.join(TEST_DATA_DIR, "sample_dir")
+    df = read_aggregate_answers_stream(study_dir)
+    assert len(df["beiwe_id"].unique()) == 2
+    assert df.shape[1] == 11
+
+
+def test_responses_by_submission():
+    study_dir = os.path.join(TEST_DATA_DIR, "sample_dir")
+    agg_data = aggregate_surveys_no_config(study_dir, study_tz="UTC")
+    surveys_dict = responses_by_submission(agg_data)
+    assert len(surveys_dict.keys()) == 1
+    assert surveys_dict["hkmxse2N7aMGfNyVMQDiWWEP"].shape[0] == 10
+
+
+def test_by_responses_by_submission_with_config():
+    study_dir = os.path.join(TEST_DATA_DIR, "sample_dir")
+    survey_settings_path = os.path.join(
+        TEST_DATA_DIR, "sample_study_surveys_and_settings.json"
+    )
+    agg_data = aggregate_surveys_config(study_dir, survey_settings_path, "UTC")
+    surveys_dict = responses_by_submission(agg_data)
+    assert len(surveys_dict.keys()) == 1
+    assert surveys_dict["hkmxse2N7aMGfNyVMQDiWWEP"].shape[0] == 10
+
+
+def test_read_empty_dir():
```

Comment:
The test function name should include the name of function under test, so: `test_aggregate_surveys_config_empty_dir()`?